docker-compose.yml

```diff
@@ -35,65 +35,81 @@ ldap:
+  container_name: gitlab-postgresql
   restart: always
-  image: mysql:5.6.25
+  image: postgres:9.6.9-alpine
   net: ${CUSTOM_NETWORK_NAME}
   expose:
-    - "3306"
+    - "5432"
   environment:
-    MYSQL_ROOT_PASSWORD: ${PASSWORD_SQL}
-    MYSQL_USER: ${GERRIT_MYSQL_USER}
-    MYSQL_PASSWORD: ${GERRIT_MYSQL_PASSWORD}
-    MYSQL_DATABASE: ${GERRIT_MYSQL_DATABASE}
+    POSTGRES_USER: ${GITLAB_POSTGRES_USER}
+    POSTGRES_PASSWORD: ${GITLAB_POSTGRES_PASSWORD}
+    POSTGRES_DB: ${GITLAB_POSTGRES_DB}
 
-gerrit:
-  container_name: gerrit
+gitlab:
+  container_name: gitlab
   restart: always
-  #build: ../images/docker-gerrit/
-  image: accenture/adop-gerrit:0.1.3
-  privileged: true
+  image: accenture/adop-gitlab:0.1.1
+  hostname: gitlab
+  links:
+    - gitlab-postgresql:postgresql
+    - gitlab-redis:redis
   net: ${CUSTOM_NETWORK_NAME}
   expose:
-    - "8080"
-    - "29418"
+    - "80"
+    - "22"
   environment:
-    - "dependency:container==gerrit-mysql"
-    - "REPO_PATH=/var/git/repos"
-    - "WEBURL=${PROTO}://${TARGET_HOST}/gerrit/"
-    - "DATABASE_TYPE=mysql"
-    - "DB_HOSTNAME=gerrit-mysql"
-    - "DB_PORT=3306"
-    - "DB_NAME=${GERRIT_MYSQL_DATABASE}"
-    - "DB_USER=${GERRIT_MYSQL_USER}"
-    - "DB_PASSWORD=${GERRIT_MYSQL_PASSWORD}"
-    - "AUTH_LOGOUTURL=${PROTO}://${TARGET_HOST}/gerrit/"
-    - "AUTH_TYPE=LDAP"
-    - "GERRIT_PASSWORD=${PASSWORD_GERRIT}"
-    - "JENKINS_PASSWORD=${PASSWORD_JENKINS}"
-    - "INITIAL_ADMIN_USER=${INITIAL_ADMIN_USER}"
-    - "INITIAL_ADMIN_PASSWORD=${INITIAL_ADMIN_PASSWORD_PLAIN}"
-    - "LDAP_SERVER=${LDAP_SERVER}"
-    - "LDAP_ACCOUNTBASE=${LDAP_FULL_DOMAIN}"
-    - "LDAP_ACCOUNTPATTERN=${LDAP_ACCOUNTPATTERN}"
-    - "LDAP_ACCOUNTFULLNAME=${LDAP_ACCOUNTFULLNAME}"
-    - "LDAP_ACCOUNTEMAILADDRESS=mail"
-    - "LDAP_USERNAME=${LDAP_ADMIN},${LDAP_FULL_DOMAIN}"
-    - "LDAP_PASSWORD=${LDAP_PWD}"
-    - "LDAP_GROUPBASE=${LDAP_GROUP_BASE_DN},${LDAP_FULL_DOMAIN}"
-    - "LDAP_GROUPPATTERN=${LDAP_GROUPPATTERN}"
-    - "LDAP_GROUPMEMBERPATTERN=${LDAP_GROUPMEMBERPATTERN}"
-    - "HTTPD_LISTENURL=proxy-${PROTO}://0.0.0.0:8080/gerrit"
-    - "USER_NAME=${GERRIT_USER_NAME}"
-    - "USER_EMAIL=${GERRIT_USER_EMAIL}"
-    - "DOWNLOAD_SCHEME=http"
+    INITIAL_ADMIN_USER: ${INITIAL_ADMIN_USER}
+    INITIAL_ADMIN_PASSWORD: ${INITIAL_ADMIN_PASSWORD_PLAIN}
+    JENKINS_USER: ${GITLAB_JENKINS_USERNAME}
+    JENKINS_PASSWORD: ${PASSWORD_JENKINS}
+    GITLAB_OMNIBUS_CONFIG: |
+     external_url "${PROTO}://${TARGET_HOST}/gitlab"
+     postgresql['enable'] = false
+     gitlab_rails['db_username'] = "${GITLAB_POSTGRES_USER}"
+     gitlab_rails['db_password'] = "${GITLAB_POSTGRES_PASSWORD}"
+     gitlab_rails['db_host'] = "gitlab-postgresql"
+     gitlab_rails['db_port'] = "5432"
+     gitlab_rails['db_database'] = "${GITLAB_POSTGRES_DB}"
+     gitlab_rails['db_adapter'] = 'postgresql'
+     gitlab_rails['db_encoding'] = 'utf8'
+     gitlab_rails['monitoring_whitelist'] = ['127.0.0.0/8', '172.0.0.0/8']
```

Comment:
The docker network (local_network) is created with 172.19.0.0/16 address space, but this does not stop CUSTOM_NETWORK_NAME been overridden and a user using a pre-defined one.

I feel we have 3 options here:
* Whitelist the whole private IP address space and loopbacks
* Pass the docker network subnet CIDR in as a parameter
* Whitelist all and rely on the proxy / Docker SDN

Reply:
As discussed with @marisbahtins, updated PR with the second option: Pass the docker network subnet CIDR in as a parameter